File: dto/admin_dto.go
package dto

import (
	"time"

	"doctor-service/entities"

	"github.com/google/uuid"
)

type AdminDoctorListItem struct {
	PublicID            uuid.UUID `json:"public_id"`
	DoctorName          string    `json:"doctor_name"`
	Email               string    `json:"email"`
	Mobile              string    `json:"mobile"`
	Status              string    `json:"status"`
	FraudScore          int       `json:"fraud_score"`
	RiskCategory        string    `json:"risk_category"`
	AssignedAdminID     *uuid.UUID `json:"assigned_admin_id,omitempty"`
	PrescriptionEnabled bool      `json:"prescription_enabled"`
	UnresolvedFlagsCount int       `json:"unresolved_flags_count"`
	CreatedAt           time.Time `json:"created_at"`
}

type AdminDashboardResponse struct {
	TotalDoctors int                   `json:"total_doctors"`
	Page         int                   `json:"page"`
	PageSize     int                   `json:"page_size"`
	TotalPages   int                   `json:"total_pages"`
	Doctors      []AdminDoctorListItem `json:"doctors"`
}

type SideBySideComparisonRow struct {
	Field          string  `json:"field"`
	DoctorEntered string  `json:"doctor_entered"`
	OCRExtracted   string  `json:"ocr_extracted"`
	MatchScore     float64 `json:"match_score"`
	IsMatch        bool    `json:"is_match"`
}

type DoctorVerificationDetailResponse struct {
	Profile              DoctorProfileDTO            `json:"profile"`
	AssignedAdminID      *uuid.UUID                  `json:"assigned_admin_id,omitempty"`
	AssignedAt           *time.Time                  `json:"assigned_at,omitempty"`
	PrescriptionEnabled  bool                        `json:"prescription_enabled"`
	Licenses             []LicenseResponse           `json:"licenses"`
	Qualifications       []QualificationResponse     `json:"qualifications"`
	Clinics              []ClinicResponse            `json:"clinics"`
	Documents            []DocumentUploadResponse    `json:"documents"`
	OCRResults           []entities.DocumentOCRResult `json:"ocr_results"`
	SideBySideComparison []SideBySideComparisonRow   `json:"side_by_side_comparison"`
	FraudScore           int                         `json:"fraud_score"`
	RiskCategory         string                      `json:"risk_category"`
	Flags                []entities.VerificationFlag `json:"flags"`
	Timeline             []entities.VerificationHistory `json:"timeline"`
	AdminActions         []entities.AdminAction      `json:"admin_actions"`
	Notes                []entities.DoctorNote       `json:"notes"`
}

type ApproveDoctorRequest struct {
	Reason string `json:"reason"`
}

type RejectDoctorRequest struct {
	Reason string `json:"reason" binding:"required"`
}

type RequestDocumentsRequest struct {
	RequiredDocuments []string `json:"required_documents" binding:"required"`
	Message           string   `json:"message" binding:"required"`
}

type AddNoteRequest struct {
	Note       string `json:"note" binding:"required"`
	Visibility string `json:"visibility"` // "INTERNAL", "VISIBLE_TO_DOCTOR"
}

File: python_backend/dto/admin_dto.py
from pydantic import BaseModel, Field
from typing import Optional, List
from uuid import UUID
from datetime import datetime

from .auth_dto import DoctorProfileDTO
from .profile_dto import (
    LicenseResponse,
    QualificationResponse,
    ClinicResponse,
    DocumentUploadResponse,
)

# Assuming these entities will be available in the python_backend.entities module
try:
    from entities import (
        DocumentOCRResult,
        VerificationFlag,
        VerificationHistory,
        AdminAction,
        DoctorNote,
    )
except ImportError:
    # Fallback types if entities are not yet translated
    from typing import Any
    DocumentOCRResult = Any
    VerificationFlag = Any
    VerificationHistory = Any
    AdminAction = Any
    DoctorNote = Any

class AdminDoctorListItem(BaseModel):
    public_id: UUID
    doctor_name: str
    email: str
    mobile: str
    status: str
    fraud_score: int
    risk_category: str
    assigned_admin_id: Optional[UUID] = None
    prescription_enabled: bool
    unresolved_flags_count: int
    created_at: datetime

class AdminDashboardResponse(BaseModel):
    total_doctors: int
    page: int
    page_size: int
    total_pages: int
    doctors: List[AdminDoctorListItem]

class SideBySideComparisonRow(BaseModel):
    field: str
    doctor_entered: str
    ocr_extracted: str
    match_score: float
    is_match: bool

class DoctorVerificationDetailResponse(BaseModel):
    profile: DoctorProfileDTO
    assigned_admin_id: Optional[UUID] = None
    assigned_at: Optional[datetime] = None
    prescription_enabled: bool
    licenses: List[LicenseResponse]
    qualifications: List[QualificationResponse]
    clinics: List[ClinicResponse]
    documents: List[DocumentUploadResponse]
    ocr_results: List[DocumentOCRResult]
    side_by_side_comparison: List[SideBySideComparisonRow]
    fraud_score: int
    risk_category: str
    flags: List[VerificationFlag]
    timeline: List[VerificationHistory]
    admin_actions: List[AdminAction]
    notes: List[DoctorNote]

class ApproveDoctorRequest(BaseModel):
    reason: Optional[str] = None

class RejectDoctorRequest(BaseModel):
    reason: str = Field(...)

class RequestDocumentsRequest(BaseModel):
    required_documents: List[str] = Field(...)
    message: str = Field(...)

class AddNoteRequest(BaseModel):
    note: str = Field(...)
    visibility: Optional[str] = Field(None, description='"INTERNAL", "VISIBLE_TO_DOCTOR"')
